components/donate/form/DonateForm.tsx

```diff
@@ -0,0 +1,97 @@
+'use client'
+
+import { useActionState } from 'react'
+import Amount from './Sum'
+import TextInput from './TextInput'
+import VisibilitySelection from './Visibility'
+import Payment from './Payment'
+import { Donation } from '@models/donate'
+import { createPaymentAction, type PaymentResponse } from '@lib/barsborsen/payment'
+import { DonationPreview } from '@components/donate/form/DonationPreview'
+import ErrorDialog from '@components/donate/ErrorDialog'
+
+export const FailedToFetchProvidersError = () => (
+  <ErrorDialog>
+    Ett okänt fel inträffade och betalningsmetoderna kunde inte hämtas. Var vänlig att dubbelkolla donationsuppgifterna.
+    Om det inte hjälper, kontakta oss på{' '}
+    <a className="link-text" href="mailto:funchef@tf.fi">
+      funchef@tf.fi
+    </a>
+  </ErrorDialog>
+)
+
+const EditDonation = ({ donation, action }: { donation?: Donation; action: (fd: FormData) => void }) => {
+  return (
+    <form action={action}>
+      <fieldset className="border border-darkgray not-prose p-4 pb-8 flex flex-col">
+        <legend>
+          <span className="px-2 text-lg font-bold uppercase">Donationsformulär</span>
+        </legend>
+        <TextInput
+          label="För- och efternamn"
+          id="donate-name"
+          name="donate-name"
+          placeholder="Svakar Teknolog"
+          defaultValue={donation?.name}
+          required
+        />
+        <TextInput
+          label="E-post"
+          id="donate-email"
+          name="donate-email"
+          type="email"
+          placeholder="svakar@teknolog.fi"
+          defaultValue={donation?.email}
+          required
+        />
+        <VisibilitySelection visibility={donation?.visibility} />
+        <Amount defaultValue={donation?.amount as number | undefined} />
+        <button className="text-white bg-darkgray hover:bg-teknologröd transition-colors rounded-2xl mt-8 py-1.5">
+          Välj betalningsmetod
+        </button>
+        <div className="flex flex-col text-sm pt-8 gap-4">
+          <p>
+            Läs mer om hantering av personuppgifter{' '}
+            <a className="link" href="https://vision.tf.fi/registerbeskrivning">
```

Comment:
Broken link, this page should be added. "Someone else" should fix but would be good to have it working before it is launched (or asap after)

Reply:
Modified to load this from Strapi in ef985a1.